### DISCUSSION TITLE
How access a keychain's data item from background state on iOS?

### DISCUSSION THREAD

**Discussion by juangarciadev** [russhwolf/multiplatform-settings#109] [Q&A]
I have implemented a background process that it is doing a request and I need to access the token stored on keychain from background app state? Do `multiplatform-settings` support to do something like this or I need to implement my own encryption to get the token?

**Comment by russhwolf** [comment]
You can use `KeychainSettings` to access data on the iOS keychain without needing to implement your own encryption. But I've never tested it from a background process.

**Reply by rahuldange1989** [reply]
Hello @russhwolf. When we try to access KeychainSettings from background process when iOS device is locked then it gives error.
 
![IMG_1063](https://github.com/user-attachments/assets/f0daa6db-f034-4805-8bdf-bd0e15b3e748)
